src/Knoop.hpp

```diff
@@ -3,18 +3,14 @@
 
 #include <map>
 #include <variant>
+#include <list>
 
 #include <range/v3/all.hpp>
 
-#ifdef USING_CPP17
-  #include <optional>
-#else 
-  #include <experimental/optional>
+#include <experimental/optional>
+namespace std { using namespace std::experimental; }
 
-  namespace std {
-    using namespace std::experimental;
-  }
-#endif
+#include "value-ptr.hpp"
 
 namespace njoy {
 namespace Knoop {
```

Comment:
You were right about the case-ness of this namespace. It should be changed before we get too far along.

Reply:
next pull request

Reply:
I saw that after I made the comment.